README getting-started build fails: -Dlocal -DskipTests skips serdes modules but app still requires them as test dependencies

## Description

**Registry Version**: `main` / `3.3.1-SNAPSHOT` (fresh clone from https://github.com/Apicurio/apicurio-registry)  
**Persistence type**: N/A — build fails before Registry starts

The README **Getting started (APIs)** section instructs new contributors to run:

```bash
./mvnw clean install -Dlocal -DskipTests
```

On a **fresh clone**, this build fails at the `apicurio-registry-app` module with a `DependencyResolutionException`.

**Root cause:** The `-Dlocal` Maven profile deactivates the `non-essential` profile, which excludes serdes and utility modules from the reactor. However, `app/pom.xml` still declares those same modules as **test-scoped** dependencies. Maven resolves test dependencies even when tests are skipped (`-DskipTests`), so the artifacts are never built or installed and the build fails.

This blocks the documented quick-start path before `quarkus:dev` can be run.

### Environment

- **OS**: Windows 10 (10.0.26200)
- **Java**: OpenJDK 21.0.6
- **Maven**: Project wrapper (`mvnw.cmd`, Apache Maven 3.3.4)
- **Clone**: Fresh fork/clone with no prior local `.m2` artifacts for serdes modules
- **Interaction**: Local development build (not runtime Registry usage)
- **Command**: `.\mvnw.cmd install -Dlocal -DskipTests -pl app -am` (also affects full `./mvnw clean install -Dlocal -DskipTests`)

**Relevant source files:**
- `README.md` (lines 58–64, 90–98) — recommends `-Dlocal -DskipTests`
- `pom.xml` (lines 1318–1342) — `non-essential` profile activates when `!local`, excludes serdes/maven-plugin/raml-test-micro-service
- `app/pom.xml` (lines 276–377) — test-scoped deps on excluded modules

### Steps to Reproduce

1. Clone Apicurio Registry from `main`:
   ```bash
   git clone https://github.com/Apicurio/apicurio-registry.git
   cd apicurio-registry
   ```

2. Follow README quick-start (Local tier):
   ```bash
   ./mvnw clean install -Dlocal -DskipTests
   ```

3. Wait until the reactor reaches `apicurio-registry-app` (~23 min on first run).

4. Observe build failure with unresolved test-scoped dependencies.

**Minimal repro (app + dependencies only):**
```bash
./mvnw install -Dlocal -DskipTests -pl app -am
```

### Expected vs Actual Behaviour

**Expected (per README):**
- `./mvnw clean install -Dlocal -DskipTests` completes successfully
- Developer can run:
  ```bash
  cd app/
  ../mvnw quarkus:dev
  ```
- REST APIs available at http://localhost:8080/apis

**Actual:**
- Build fails at `apicurio-registry-app` with `DependencyResolutionException`
- Six test-scoped dependencies cannot be resolved because their modules were excluded from the `-Dlocal` reactor
- Server never starts; `quarkus:dev` cannot be run without a successful install

**Workaround (confirmed working):**
```bash
# Option A: Default tier (no -Dlocal)
./mvnw clean install -DskipTests

# Option B: Build serdes separately first, then resume app
./mvnw install -DskipTests -pl serdes/kafka/avro-serde,serdes/kafka/protobuf-serde,serdes/kafka/jsonschema-serde,serdes/pulsar/avro-serde,utils/maven-plugin,utils/raml-test-micro-service -am
./mvnw install -DskipTests -rf :apicurio-registry-app
```

### Logs

```
[INFO] Reactor Summary for Registry 3.3.1-SNAPSHOT:
[INFO] Registry :: App .................................... FAILURE [12:07 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project apicurio-registry-app: Could not resolve dependencies for project io.apicurio:apicurio-registry-app:jar:3.3.1-SNAPSHOT
[ERROR] dependency: io.apicurio:apicurio-registry-avro-serde-kafka:jar:3.3.1-SNAPSHOT (test)
[ERROR]   Could not find artifact io.apicurio:apicurio-registry-avro-serde-kafka:jar:3.3.1-SNAPSHOT
[ERROR] dependency: io.apicurio:apicurio-registry-protobuf-serde-kafka:jar:3.3.1-SNAPSHOT (test)
[ERROR]   Could not find artifact io.apicurio:apicurio-registry-protobuf-serde-kafka:jar:3.3.1-SNAPSHOT
[ERROR] dependency: io.apicurio:apicurio-registry-jsonschema-serde-kafka:jar:3.3.1-SNAPSHOT (test)
[ERROR]   Could not find artifact io.apicurio:apicurio-registry-jsonschema-serde-kafka:jar:3.3.1-SNAPSHOT
[ERROR] dependency: io.apicurio:apicurio-registry-avro-serde-pulsar:jar:3.3.1-SNAPSHOT (test)
[ERROR]   Could not find artifact io.apicurio:apicurio-registry-avro-serde-pulsar:jar:3.3.1-SNAPSHOT
[ERROR] dependency: io.apicurio:apicurio-registry-maven-plugin:jar:3.3.1-SNAPSHOT (test)
[ERROR]   Could not find artifact io.apicurio:apicurio-registry-maven-plugin:jar:3.3.1-SNAPSHOT
[ERROR] dependency: raml-test-micro-service:jar:3.3.1-SNAPSHOT (test)
[ERROR]   Could not find artifact raml-test-micro-service:jar:3.3.1-SNAPSHOT
```

Build finished: `2026-07-10T21:03:48+05:30`, total time: 22:57 min

--- a/app/pom.xml
+++ b/app/pom.xml
@@ -274,11 +274,6 @@
     </dependency>
     <dependency>
       <groupId>io.apicurio</groupId>
-      <artifactId>apicurio-registry-avro-serde-kafka</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>io.apicurio</groupId>
       <artifactId>apicurio-registry-java-sdk</artifactId>
       <scope>test</scope>
     </dependency>
@@ -295,27 +290,6 @@
     <dependency>
       <groupId>io.kiota</groupId>
       <artifactId>kiota-http-vertx</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>io.apicurio</groupId>
-      <artifactId>apicurio-registry-protobuf-serde-kafka</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>io.apicurio</groupId>
-      <artifactId>apicurio-registry-jsonschema-serde-kafka</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>io.apicurio</groupId>
-      <artifactId>apicurio-registry-avro-serde-pulsar</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>io.apicurio</groupId>
-      <artifactId>apicurio-registry-maven-plugin</artifactId>
-      <type>maven-plugin</type>
       <scope>test</scope>
     </dependency>
     <dependency>
@@ -367,12 +341,6 @@
     <dependency>
       <groupId>org.awaitility</groupId>
       <artifactId>awaitility</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>${project.groupId}</groupId>
-      <artifactId>raml-test-micro-service</artifactId>
-      <version>${project.version}</version>
       <scope>test</scope>
     </dependency>
     <dependency>
@@ -716,5 +684,53 @@
       </plugin>
     </plugins>
   </build>
+
+  <!-- Test-scoped dependencies on modules excluded from the reactor by the -Dlocal profile
+       (serdes, maven-plugin, raml-test-micro-service). Keep them out of -Dlocal builds so
+       dependency resolution succeeds even with -DskipTests; include them for non-local test runs. -->
+  <profiles>
+    <profile>
+      <id>local-excluded-test-deps</id>
+      <activation>
+        <property>
+          <name>!local</name>
+        </property>
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>io.apicurio</groupId>
+          <artifactId>apicurio-registry-avro-serde-kafka</artifactId>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>io.apicurio</groupId>
+          <artifactId>apicurio-registry-protobuf-serde-kafka</artifactId>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>io.apicurio</groupId>
+          <artifactId>apicurio-registry-jsonschema-serde-kafka</artifactId>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>io.apicurio</groupId>
+          <artifactId>apicurio-registry-avro-serde-pulsar</artifactId>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>io.apicurio</groupId>
+          <artifactId>apicurio-registry-maven-plugin</artifactId>
+          <type>maven-plugin</type>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>${project.groupId}</groupId>
+          <artifactId>raml-test-micro-service</artifactId>
+          <version>${project.version}</version>
+          <scope>test</scope>
+        </dependency>
+      </dependencies>
+    </profile>
+  </profiles>
 
 </project>

